File: FrontEnd/src/app/shared/models/Navbar.ts
export class Navbar{
    public message: NavbarMessage;
    public errorHappen: boolean;
}

export class NavbarMessage {
    public level0: NavbarElement[];
    public level1: NavbarElement[];
}

export class NavbarElement{
    public id: number;
    public Name: string;
    public Icon: string;
    public parentId: number;
    public levelInMenu: number;
    public link: string;
    public abilityId: number;
}

File: BackEnd/Models/Domain/Navbar.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Models.Domain
{
    public class Navbar
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Icon { get; set; }
        public int LevelInMenu { get; set; }
        public int Order { get; set; }

        public int? AbilityId { get; set; }
        public Ability Ability { get; set; }
        public int? ParentId { get; set; }
        public Navbar Parent { get; set; }
    }
}
